### ISSUE TITLE
Fix discretizing methods to work with ContinuousFactor

### ISSUE THREAD

**Issue by ankurankan** [pgmpy/pgmpy#746]
Were originally written to work with `ContinousNode` but now since it has been replace with `ContinuousFactor`, we need to update it accordingly.

**Comment by ankurankan** [comment]
Related #742

**Comment by yashu-seth** [comment]
@ankurankan The discretization algorithms that I used worked only for uni-dimensional cases. One solution is to modify them for multivariate cases. Or we can look for multivariate discretization algorithms. What do you think?

**Comment by ankurankan** [comment]
@yashu-seth Yes, I did find some algorithms which would be good for pgmpy but don't remember the papers. I will add those if I find them.

**Comment by kris-singh** [comment]
@ankurankan can you post these papers here.
